saleor/discount/models.py

```diff
@@ -119,6 +119,8 @@ def get_discount(self, channel: Channel):
     def get_discount_amount_for(self, price: Money, channel: Channel):
         discount = self.get_discount(channel)
         after_discount = discount(price)
+        if type(after_discount) is TaxedMoney:
```

Comment:
See typing for this method. If you're receiving here TaxedMoney, I would say that there is an incorrect call of this method as typing expects to get normal Money price.